pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -84,7 +84,7 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 		}
 		// retrieve the pod object from pod informer with the namespace key
 		informerPodObj, exists, err := dswp.podInformer.GetStore().GetByKey(dswPodKey)
-		if err != nil || informerPodObj == nil {
+		if err != nil || (exists && informerPodObj == nil) {
```

Comment:
@saad-ali DO we want to make this check more descriptive? Is one worth logging more than another?

Trying to understand:

if err != nil {
   glog.Errorf("podInformer GetBy key failed with an error....")
  continue
}

if (exists && informerPodObj == nil) {
    glog.Info("podInformer GetByKey found pod that is already deleted %q", dwsPodKey)
   continue
}